### PR TITLE
build: make ci testing conditional on engines field in package.json, move configs to Node 12

### PR DESCRIPTION
**Pull request by sofisl**
Deleting Node 10 configs here, but we will also have to delete them in each repo.

cc @bcoe

### CODE REVIEW

**Review by bcoe — CHANGES_REQUESTED** [review]
Looking slick to me, left a recommendation for a test and a tiny refactor.

File: synthtool/languages/node.py [hunk 1]
```diff
@@ -48,6 +48,7 @@ def read_metadata():
         data["repository"] = f'{repo["owner"]}/{repo["name"]}'
         data["repository_name"] = repo["name"]
         data["lib_install_cmd"] = f'npm install {data["name"]}'
+        data["engine"] = data["engines"]["node"].replace(">=", "").replace(".0", "")
```
[the remark this answers is not in the document]
Comment (sofisl) [reply synthtool/languages/node.py]: cool, PTAL!

File: synthtool/gcp/templates/node_library/.github/workflows/ci.yaml [hunk 2]
```diff
@@ -9,7 +9,7 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        node: [10, 12, 14, 16]
+        node: [{{metadata['engine'] | int}}, {{metadata['engine'] | int+2}}, {{metadata['engine'] | int+4}}, {{metadata['engine'] | int+6}}]
```
Comment (bcoe) [inline synthtool/gcp/templates/node_library/.github/workflows/ci.yaml:12]: Slight problem, node 18 won't exist for another 20 days.

Let's just make this `{{metadata['engine'] | int}}, {{metadata['engine'] | int+2}}, {{metadata['engine'] | int+4}}` for now, and in a couple months we can add Node 18.

File: tests/test_node.py [hunk 3]
```diff
@@ -43,6 +43,13 @@ def test_quickstart_metadata_with_snippet():
         assert "samples/quickstart.js" in sample_names
 
 
+def test_metadata_engines_field():
+    with util.chdir(FIXTURES / "node_templates" / "standard"):
+        metadata = node.template_metadata()
+
```
Comment (bcoe) [inline tests/test_node.py:49]:
```suggestion
```

File: synthtool/gcp/templates/node_library/.github/workflows/ci.yaml (hunk 2 above)
Comment (bcoe) [reply synthtool/gcp/templates/node_library/.github/workflows/ci.yaml:12]:
```suggestion
        node: [{{metadata['engine'] | int}}, {{metadata['engine'] | int+2}}, {{metadata['engine'] | int+4}}]
```

File: tests/fixtures/node_templates/standard/package.json [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+      "sinon": "^10.0.0",
+      "ts-loader": "^8.0.0",
+      "typescript": "^3.8.3",
+      "webpack": "^5.0.0",
+      "webpack-cli": "^4.0.0",
+      "@microsoft/api-documenter": "^7.8.10",
+      "@microsoft/api-extractor": "^7.8.10"
+    }
+  }
+
```
Comment (bcoe) [inline tests/fixtures/node_templates/standard/package.json:73]:
```suggestion
```

**Review by bcoe — APPROVED** [review]

**Pull request merged**